### PR TITLE
Refactor for ec2removeimage script

### PR DESCRIPTION
**Pull request by apozsuse**
- Functionality in the script has been encapsulated into functions
- Added tests to verify argument/config management and script functionality (mocking some boto3 response)

### CODE REVIEW

**Review by smarlowucf — APPROVED** [review]

File: ec2removeimg [hunk 1]
```diff
@@ -28,209 +28,282 @@ from ec2imgutils.ec2imgutilsExceptions import (
...
-    config = utils.get_config(config_file)
-except Exception as e:
-    logger.error(e)
-    sys.exit(1)
-
-access_key = args.access_key
-if not access_key:
+# ----------------------------------------------------------------------------
+def parse_args(args):
+    """Commang argument parsing function."""
```
Comment (rjschwei) [inline ec2removeimg:34]: s/Commang/Command/

File: ec2removeimg [hunk 2]
```diff
@@ -28,209 +28,282 @@ from ec2imgutils.ec2imgutilsExceptions import (
...
+        help='Program version'
+    )
+
+    parsed_args = parser.parse_args(args)
+    return parsed_args
+
+
+# ----------------------------------------------------------------------------
+def check_required_arguments(args, logger):
+    """This function is used to assure the additional requirements over
```
Comment (rjschwei) [inline ec2removeimg:153]: s/assure/ensure/

File: ec2removeimg [hunk 3]
```diff
@@ -28,209 +28,282 @@ from ec2imgutils.ec2imgutilsExceptions import (
...
+        error_msg = 'ec2removeimg: error: one of the arguments '
+        error_msg += '--image-id --image-name --image-name-frag '
+        error_msg += '--image-name-match is required'
+        logger.error(error_msg)
+        sys.exit(1)
+
+
+# ----------------------------------------------------------------------------
+def get_config(args, logger):
+    """Function to the the configutation parsed from the configuration file
```
Comment (rjschwei) [inline ec2removeimg:179]: "the the" maybe "get the" ?

**Review by rjschwei — CHANGES_REQUESTED** [review]

File: ec2removeimg (hunk 1 above)
Comment (apozsuse) [reply ec2removeimg]: done

**Review by rjschwei — APPROVED** [review]

**Pull request merged**